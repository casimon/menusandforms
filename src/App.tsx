import React from 'react'
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Redirect,
} from 'react-router-dom'

import Form from './screens/Form'
import Menu from './screens/Menu'
import PageNotFound from './screens/PageNotFound'

function App() {
  return (
    <Router>
      <Switch>
        <Route exact path='/'>
          <Redirect to='/menu/1' />
        </Route>
        <Route exact path='/menu/:id' component={Menu} />
        <Route exact path='/form/:id' component={Form} />
        <Route exact path='*' component={PageNotFound} status={404} />
      </Switch>
    </Router>
  )
}

export default App
