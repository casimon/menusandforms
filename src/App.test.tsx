import React from 'react'

import { render } from '@testing-library/react'
import App from './App'

const init = () => render(<App />)

describe('My App component', () => {
  test('should render app without error ', () => {
    const { container } = init()
    expect(container).toMatchSnapshot()
  })
})
