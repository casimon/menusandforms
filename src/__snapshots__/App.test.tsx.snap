// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My App component should render app without error  1`] = `
<div>
  <div
    class="makeStyles-root-1"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthSm"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-direction-xs-column MuiGrid-justify-xs-center"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <h1
            class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
          >
            Want to win $5000 with Afterpay?
          </h1>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <a
            aria-disabled="false"
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
            href="/form/1"
            tabindex="0"
          >
            <span
              class="MuiButton-label"
            >
              Hell yes!
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </a>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <a
            aria-disabled="false"
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
            href="/menu/2"
            tabindex="0"
          >
            <span
              class="MuiButton-label"
            >
              Maybe
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </a>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <a
            aria-disabled="false"
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
            href="/nope"
            tabindex="0"
          >
            <span
              class="MuiButton-label"
            >
              Nope
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
